app/main.py

```diff
@@ -1,3 +1,58 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        if km < 0:
+            raise ValueError("Must be a positive number")
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, num: int) -> Distance:
+        return Distance(self.km * num)
+
+    def __truediv__(self, num: Union[int, float]) -> Distance:
+        return Distance(round(self.km / num, 2))
+
+    def __lt__(self, other: Union[Distance, int, float]) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        return self.km < other
+
+    def __gt__(self, other: Union[Distance, int, float]) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        return self.km > other
+
+    def __eq__(self, other: Union[Distance, int, float]) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __le__(self, other: Union[Distance, int, float]) -> bool:
+        if isinstance(other, Distance):
+            return not (self.km > other.km)
+        return not (self.km > other)
```

Comment:
since you defined __lt__ and __eq__ methods above you could simplify this to one line:
return not self > other